### PR TITLE
feat(#545) publish cht-app-ide docker image

### PR DESCRIPTION
**Pull request by jkuester**
# Description

Adds GitHub actions workflow to build and publish the `cht-app-ide` image every time a tag is pushed.

I have based the workflow configuration off of the [official documentation](https://docs.github.com/en/actions/deployment/deploying-to-your-cloud-provider/deploying-to-amazon-elastic-container-service).

Note that we should not merge this PR until https://github.com/medic/medic-infrastructure/issues/570 is complete and the credentials have been configured.

Closes https://github.com/medic/cht-conf/issues/545

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or integration tests where appropriate
- Backwards compatible: Works with existing data and configuration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by jkuester** [comment]
@dianabarsan can you have a look at this when you get the chance?  There are two things I am wondering if you know of from experience with the other cht Docker images:

- Do we need to worry about tagging a `latest` image, or does Amazon ECR handle that automatically?
- Do you know of any convenient way to test this workflow config before we commit it?  (Normally I would just make a fork and tinker with things there, but I don't really have any ECR logins handy....)

File: devcontainer.cht-app-ide/.devcontainer.json [hunk 1]
```diff
@@ -1,5 +1,5 @@
 {
-  "image": "cht-ide",
+  "image": "public.ecr.aws/s5s3h4s7/cht-app-ide:latest",
```
Comment (dianabarsan) [inline devcontainer.cht-app-ide/.devcontainer.json:2]: I _think_ that you can use the `medic` alias here: https://github.com/medic/cht-core/issues/7952

File: README.md [hunk 2]
```diff
@@ -88,12 +78,12 @@ To open a terminal running on you _host environment_ in VSCode, open the Command
...
 ##### Standalone Docker utility
 
-If you are not using VSCode, you can use the Docker image as a standalone utility from the command line.  Instead of using the `cht ...` command, you can run `docker run -it --rm -v "$PWD":/workdir cht-ide ...`. This will create an ephemeral container with access to your current directory that will run the given cht command. (Do not include the `cht` part of the command, just your desired actions/parameters.)
+If you are not using VSCode, you can use the Docker image as a standalone utility from the command line.  Instead of using the `cht ...` command, you can run `docker run -it --rm -v "$PWD":/workdir cht-app-ide ...`. This will create an ephemeral container with access to your current directory that will run the given cht command. (Do not include the `cht` part of the command, just your desired actions/parameters.)
 
 Run the following command inside the project directory to bootstrap your new CHT project:
 
 ```shell
-docker run -it --rm -v "$PWD":/workdir cht-ide initialise-project-layout
+docker run -it --rm -v "$PWD":/workdir public.ecr.aws/s5s3h4s7/cht-app-ide initialise-project-layout
```
Comment (dianabarsan) [inline README.md:86]: I think that you can use the medic alias here: https://github.com/medic/cht-core/issues/7952

**Review by dianabarsan — APPROVED** [review]
Very nice!

**Comment by jkuester** [comment]
Based on the conversation in https://github.com/medic/medic-infrastructure/issues/570, I have updated the workflow to push the image to the Docker Hub instead of ECR (based the workflow config off of the [official docs](https://docs.github.com/en/actions/publishing-packages/publishing-docker-images)).

**Review by dianabarsan — APPROVED** [review]
Very nice. I love just how easy the docker actions are to use.

**Pull request merged**